py/client/pydeephaven/agg.py

```diff
@@ -0,0 +1,356 @@
+        formula_param (str): the parameter name within the formula
+        cols (List[str]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(formula=_GrpcAggSpec.AggSpecFormula(formula=formula, param_token=formula_param))
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def last(cols: List[str] = None) -> Aggregation:
+    """Create a Last aggregation.
+
+    Args:
+        cols (List[str]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(last=_GrpcAggSpec.AggSpecLast())
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def min_(cols: List[str] = None) -> Aggregation:
+    """Create a Min aggregation.
+
+    Args:
+        cols (List[str]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(min=_GrpcAggSpec.AggSpecMin())
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def max_(cols: List[str] = None) -> Aggregation:
+    """Create a Max aggregation.
+
+    Args:
+        cols (List[str]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(max=_GrpcAggSpec.AggSpecMax())
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def median(cols: List[str] = None) -> Aggregation:
+    """Create a Median aggregation.
+
+    Args:
+        cols (List[str]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(median=_GrpcAggSpec.AggSpecMedian(average_evenly_divided=True))
+    return _AggregationColumns(agg_spec=agg_spec, cols=cols)
+
+
+def pct(percentile: float, cols: List[str] = None) -> Aggregation:
+    """Create a Percentile aggregation.
+
+    Args:
+        percentile (float): the percentile used for calculation
+        cols (List[str]): the column(s) to aggregate on, can be renaming expressions, i.e. "new_col = col";
+            default is None, only valid when used in Table agg_all_by operation
+
+    Returns:
+        an aggregation
+    """
+    agg_spec = _GrpcAggSpec(percentile=_GrpcAggSpec.AggSpecPercentile(percentile=percentile,
+                                                                      average_evenly_divided=True))
```

Comment:
I think `average_evenly_divided` should be a parameter for the user. *NOTE*: the default on the server is `False` and should probably default to `False` here as well.

Reply:
#3626